### PR TITLE
Divyanshu | Stage | Undefined is getting sent in url when user go to edit mode voucher page

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cww6y3z

### CODE REVIEW

**Review by dilpreetsinghofficial — DISMISSED** [review]

File: apps/web-giddh/src/app/vouchers/list/list.component.ts [hunk 1]
```diff
@@ -711,8 +711,19 @@ export class VoucherListComponent implements OnInit, OnDestroy {
...
     public showVoucherPreview(voucherUniqueName: string): void {
+        const queryParams =  { 
+            page: this.advanceFilters.page, 
+            from: this.advanceFilters.from, 
+            to: this.advanceFilters.to, 
+        };
+
+        const searchString = this.advanceFilters.q ?? this.advanceFilters.proformaNumber ?? this.advanceFilters.estimateNumber ?? this.advanceFilters.purchaseOrderNumber;
+        if (searchString?.length){
+            queryParams['search'] = searchString
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/list/list.component.ts]:
```suggestion
            queryParams['search'] = searchString;
```

File: apps/web-giddh/src/app/vouchers/preview/preview.component.ts [hunk 2]
```diff
@@ -231,14 +231,19 @@ export class VouchersPreviewComponent implements OnInit, OnDestroy {
...
+                        this.getAllVouchers();
+                    }
                 }
             }
         });
         this.isCompany = this.generalService.currentOrganizationType === OrganizationType.Company;
         this.imgPath = isElectron ? 'assets/images/' : AppUrl + APP_FOLDER + 'assets/images/';
         this.search.valueChanges.pipe(debounceTime(700), distinctUntilChanged(), takeUntil(this.destroyed$)).subscribe(search => {
-            if (search || search === '') {
+            if ((search || search === '')) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/preview/preview.component.ts]: what's the use of extra braces?

File: apps/web-giddh/src/app/vouchers/preview/preview.component.ts [hunk 3]
```diff
@@ -1127,6 +1130,11 @@ export class VouchersPreviewComponent implements OnInit, OnDestroy {
             to: this.advanceFilters.to,
             page: this.advanceFilters.page
         }
+        
+        const searchString = this.advanceFilters.q ?? this.advanceFilters.proformaNumber ?? this.advanceFilters.estimateNumber ?? this.advanceFilters.purchaseOrderNumber;
+        if (searchString?.length){
+            queryParams['search'] = searchString
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vouchers/preview/preview.component.ts]:
```suggestion
            queryParams['search'] = searchString;
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vouchers/preview/preview.component.ts (hunk 2 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vouchers/preview/preview.component.ts]: By mistake .

File: apps/web-giddh/src/app/vouchers/list/list.component.ts (hunk 1 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vouchers/list/list.component.ts]: Done

File: apps/web-giddh/src/app/vouchers/preview/preview.component.ts (hunk 3 above)
Comment (ankitdubeyoffice) [reply apps/web-giddh/src/app/vouchers/preview/preview.component.ts]: Done

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**